### PR TITLE
Gradle - Correctly merge classes dir when using dev mode

### LINKED ISSUE #42860: Quarkus output class in incorrect folder on live reload (gradle, multimodule, kotlin)
### Describe the bug

In a gradle project with :
 - a `common` module using java+kotlin+jandex
 - an `app` module using java+quarkus

Once the app started, if we modify a java source of the `common` module
And trigger a live reload
Then the java class is produced in `common/build/resources` directory rather than `common/build/classes` directory.

Once this has happened, app restart won't work with message like :
```
Execution failed for task ':common:jar'.
> Entry fr/common/Foo.class is a duplicate but no duplicate handling strategy has been set
```

### Expected behavior

On live reload recompilation, classes should be produced in `build/classes` directory ; (event if there is the kotlin plugin)

### Actual behavior

Class produced in resource folder leading to on future start :
```
Execution failed for task ':common:jar'.
> Entry fr/common/Foo.class is a duplicate but no duplicate handling strategy has been set
```

### How to Reproduce?

[quarkus-bug-common-lib.zip](https://github.com/user-attachments/files/16798128/quarkus-bug-common-lib.zip)

Using the attached zip, launch `./gradlew :app:quarkusDev`
Modifiy `common/src/main/java/fr/common/Foo.java` (even the comment line).
Open browser to `http://localhost:8080` to trigger a live reload.

See that `common/build/resources` contains file `fr/common/Foo.class`
Stop the app with Ctrl+c, then launch again `./gradlew :app:quarkusDev`
Error about duplicate resource when executing jar task (because the class file is both in build/classes and build/ressources)

### Output of `uname -a` or `ver`

Darwin MacBook-Pro-de-Yann.local 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:13:04 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

openjdk version "21.0.3" 2024-04-16 LTS OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode)

### Quarkus version or git rev

3.13.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.8

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gsmet**
We were returning an empty dir before which would trigger the use of the resources dir instead.

Also simplify the logic as it was quite hard to follow and it's actually quite simple.

Fixes #42860

### CODE REVIEW

File: devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 1]
```diff
@@ -633,7 +633,7 @@ private void addLocalProject(ResolvedDependency project, GradleDevModeLauncher.B
             }
         }
         Path classesDir = classesDirs.isEmpty() ? null
-                : QuarkusGradleUtils.mergeClassesDirs(classesDirs, project.getWorkspaceModule().getBuildDir(), root, false);
+                : QuarkusGradleUtils.mergeClassesDirs(classesDirs, project.getWorkspaceModule().getBuildDir(), true, false);
```
Comment (gsmet) [inline devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java:636]: This is the actual fix.

File: devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java [hunk 2]
```diff
@@ -46,44 +46,35 @@ public static String getClassesDir(SourceSet sourceSet, File tmpDir, boolean pop
     }
 
     public static Path mergeClassesDirs(Collection<Path> classesDirs, File tmpDir, boolean populated, boolean test) {
```
Comment (gsmet) [inline devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java:48]: I cleaned up this method because it was coded with quite a weird pattern that made it hard to understand because we initially had an iterator. Now that we have plain collection, we can use a much simpler approach.

File: devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java [hunk 3]
```diff
@@ -46,44 +46,35 @@ public static String getClassesDir(SourceSet sourceSet, File tmpDir, boolean pop
...
-            if (!Files.exists(next)) {
-                continue;
+        List<Path> existingClassesDirs = classesDirs.stream().filter(p -> Files.exists(p)).toList();
+
+        if (existingClassesDirs.size() == 0) {
+            return null;
+        }
+
+        if (existingClassesDirs.size() == 1) {
+            return existingClassesDirs.iterator().next();
```
Comment (aloubyansky) [inline devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java:56]:
```suggestion
            return existingClassesDirs.get(0);
```

**Review by aloubyansky — APPROVED** [review]

File: devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java (hunk 3 above)
Comment (gsmet) [reply devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java:56]: Ah yes, I made it a list in the end, it was initially a collection, will fix!
